src/node_worker.cc

```diff
@@ -542,7 +542,19 @@ void Worker::New(const FunctionCallbackInfo<Value>& args) {
 
 #ifndef NODE_WITHOUT_NODE_OPTIONS
     std::string node_options;
-    if (env_vars->Get("NODE_OPTIONS").To(&node_options)) {
+    std::string parent_node_options;
+    // If the worker code passes { env: { ...process.env, ... } } or
+    // the NODE_OPTIONS is otherwise copied into the env option, don't
+    // throw because the worker spawning code is not usually in charge of
+    // how the NODE_OPTIONS is configured for the parent.
+    // TODO(joyeecheung): a more intelligent filter may be more desirable.
+    // but a string comparison is good enough(TM) for the case where the
+    // worker spawning code just wants to pass the parent configuration down
+    // with some valid overrides.
+    USE(env->env_vars()->Get("NODE_OPTIONS").To(&parent_node_options));
+    if (args[1]->IsObject() &&
+        env_vars->Get("NODE_OPTIONS").To(&node_options) &&
+        node_options != parent_node_options) {
```

Comment:
Worker will ignore valid per-env options in `NODE_OPTIONS` when the process was started with a same `NODE_OPTIONS`.

```js
#!/usr/bin/env NODE_OPTIONS=--trace-exit ./out/Debug/node

const { Worker, isMainThread } = require('worker_threads')

if (isMainThread) {
  new Worker(__filename, {
    env: {
      NODE_OPTIONS: '--trace-exit'
    }
  })
} else {
  setTimeout(() => {
    process.exit(1)
  }, 1000)
}
```

Reply:
Oh, right, it still needs to be parsed to mutate the per_isolate_opts. Maybe we should just check in the error branch though that do require a pretty dumb `error.starts_with("invalid value for NODE_OPTIONS")` check for all the errors..

Reply:
..or probably just switch to `kDisallowedInEnvvar` if the NODE_OPTIONS are character-by-character equal with the parent one

Reply:
Updated to just wrap it around a throw-away PerProcessOption and added a test.